test/hamiltonian_consistency.jl

```diff
@@ -0,0 +1,60 @@
+using Test
+using DFTK
```

Comment:
I see the point, but I would really split this up into individual @testset sections *for each term*, such that you can directly see what fails if something does. I am happy to do that, but just let me know you are ok before I do.

Reply:
I don't really buy that argument: if something goes wrong, you just go there and debug. But I'm fine with adding it if you want (and if it doesn't add too much complexity)